### PR TITLE
add more instrumentation for performance tracing

### PR DESCRIPTION
**Pull request by syphar**
To potentially replace the rendering-times recording. 

Also I found another SQL query we don't need any more.

### CODE REVIEW

**Review by GuillaumeGomez — APPROVED** [review]
Looks good to me but please wait for @Nemo157's review before merging.

**Review by Nemo157 — APPROVED** [review]

**Pull request merged**
